### ISSUE TITLE
Bug Report: HTTP services with annotations over time will get into a bad state with the router

### ISSUE THREAD

**Issue by Kampe** [skupperproject/skupper#390]
Hello again!

Sorry for the strange title, I'm unsure of how else to describe what is happening and I haven't actually caught what's happening in the act as this is something we notice over time while letting skupper run (in this case for almost exactly 7 days), eventually queues will get jammed up, seemingly in just the http proxies. However this is what I have noticed about the services when the odd behavior begins happening:

We'll see the proxies when listed out show the metadata from the router itself:
```
Services exposed through Skupper:
    cloud-api (http port 5443) with targets
      => app=cloud-api,application=skupper-router,skupper.io/component=router name=cloud-api
```

The solution is to delete the service and recreate it, which will allow skupper to take over the pod's ports again.  
  
Once recreated, it'll be in working order again and will look as such:
```
Services exposed through Skupper:
  cloud-api (http port 5443) with targets
    => app=cloud-api name=cloud-api
```

Given some of the behavior I assume this may be related to the service controller, however I don't see anything out of wack in the logs aside from 
```
Failed to authenticate : read /etc/console-users: is a directory
```

Let me know if there's any clues as to where I can poke around at to give you some further details, for context we deploy skupper in a very typical cloud "hub spoke" edge pattern using the site controller.

**Comment by grs** [comment]
The annotation on the service has turned out to be a somewhat problematic mechanism. The original selector defines the target pods, but skupper then overwrites this. It tries to keep the original as an annotation, but there are ways that might be lost. 

A simpler and safer option is to annotate the deployment instead. Would that be an option for you? If so it might be worth deprecating the service annotation (except where the service is itself the target. not the pods it points at)

If not we will try and make things more robust for service annotations. (I see one obvious fix that I think would alleviate your particular case, though the mechanism remains in my view somewhat fragile).

**Comment by Kampe** [comment]
I figured that was potentially what was happening there with how skupper manages the pod ports. As far as annotating the deployments themselves that would work just fine actually, and would make a little more sense the more I think about it.  Is that currently possible with v0.4.2?  I can make that change and give it a good test.

**Comment by grs** [comment]
Yes, annotations on deployments has been possible since 0.2.

**Comment by Kampe** [comment]
Will swap and report back with any findings, is this work tracked on another issue with a better description? Want me to leave the issue open for now?

**Comment by grs** [comment]
Yes, lets leave this open for now.

**Comment by Kampe** [comment]
Will do, am I able to assign a specific name of the skuppered service to my edges or does the service controlled on edge just assume the service name from the deployment?  

The reason I ask is previously I would create entirely new special services just for the skupper routers to use, while all other "local traffic" that would need to hit the skuppered deployment would communicate on a different service, to the same ports. Also I assume this works the same on Daemonsets and StatefulSets?

**Comment by grs** [comment]
You can use the skupper.io/address annotation to choose the service name/address. Unfortunately at present we don't recognise annotations on DaemonSets or StatefulSets, but that should be easy to add. You can of course use the skupper expose command, but that is obviously not ideal for more declarative use.

**Comment by grs** [comment]
Fixed in 0.5.1
